### ISSUE TITLE
Extensive memory usage with a DVR live stream

### ISSUE THREAD

**Issue by BucherTomas** [jwplayer/jwplayer#1807]
### Expected Behavior

*When processing an HLS live stream, the player should not incrementally consume memory.*

### Actual Behavior

*The memory consumption begins to increase right after playback start. If left running for some prolonged time, it can even crash the browser tab the player is opened in with an out of memory error. It seems that DVR live streams are prone to this behavior much faster, likely due to the number of referenced chunks in the m3u8 manifest.*

### Steps to reproduce

1. Use the following URL http://pspcr-live.service.cdn.cra.cz/live.pscr/smil:snemovna2/index.m3u8?DVR in demo player http://demo.jwplayer.com/developer-tools/http-stream-tester/
2. Observe the memory consumption

Heap snapshot from Chrome after running for mere 3 minutes, but the memory usage already visibly increased.

![chromeheap](https://cloud.githubusercontent.com/assets/18445121/22789718/3a2f07ba-eee4-11e6-8a01-b7e3425bd286.png)

If JW Player includes hls.js library, I first thought that it might be caused by this issue https://github.com/dailymotion/hls.js/issues/939 but if I'm reading it right by the commits in JW Player, version 7.9.1 should already include the fixes, however when tried with https://jsfiddle.net/Lgs0ou8s/4/, the same behavior is still there.

### Environment

- *PC, MacOS*
- *Windows 7, Windows 10, Sierra 10.12.3*
- *Chrome 56.0.2924.87, Chrome Canary 58.0.3007.0, Firefox 50.1, Safari 10.0.3*
- *JW Player 7.9.0 | http://demo.jwplayer.com/developer-tools/http-stream-tester/ and JW Player 7.9.1 | https://jsfiddle.net/Lgs0ou8s/4/ with --allow-running-insecure-content flag in Chrome to circumvent mixed content error*

**Comment by BucherTomas** [comment]
Actually, Safari is working fine likely due to its native HLS support. It was one of our testing browsers and it was added by mistake among the affected ones.

**Comment by BucherTomas** [comment]
Just a heads up, this issue does not manifest itself if the version 7.9.0 is replaced with 7.8.7.

I left the aforementioned DVR stream run for 3 hours and once RAM usage stabilizes after a couple of minutes, associated Chrome process fluctuates from 300 to 400 MB, whereas version 7.9.0 crashes the browser tab in the meantime, the last value I saw was around 1.7 GB on my machine which had 4 GB of free RAM when the stream started.

**Comment by deltafactory** [comment]
Seeing similar issues. 1 hour of live stream with DVR saw browser's memory usage at 4GB. Closing tab and waiting a few seconds for GC, memory usage in the process was down to 800MB.

## Environment
* Ubuntu Desktop 16.04.1
* Firefox 51.0.1
* JW Player 7.9.1

**Comment by robwalch** [comment]
Can you reproduce the memory leak here http://video-dev.github.io/hls.js/demo/ ? It would help if we could isolate the issue to hls.js or jwplayer.

**Comment by BucherTomas** [comment]
I did test it with hls.js demo player before I wrote the ticket here and it was fine, don't recall what version it was at that time, but it was a later build including the mentioned commit that was supposed to fix this.

I tried the recent JWPlayer build 7.9.3 yesterday with the same stream in the same Chrome version 56.0.2924.87 as before and despite the fact that I don't see anything remotely related to this problem in release notes, the memory usage is stable here again as well.

@deltafactory , what's your experience with 7.9.3?
